stock_available_to_promise_release/models/stock_move.py

```diff
@@ -363,6 +369,23 @@ def _run_stock_rule(self):
 
         return True
 
+    def _release_get_deadline(self, date_planned):
+        prep_time = self.env.company.stock_release_max_prep_time
+        date_deadline = fields.Datetime.add(date_planned, minutes=prep_time)
+        return date_deadline
+
+    def _release_split_set_date(self, date_planned):
+        """Set released move planned date to pulled moves planned date"""
+        self.date = self._release_get_deadline(date_planned)
+
+    def _release_set_procurement_date(self, values, date_planned):
+        # We must set the deadline on the move before pulling to allow pulled
+        # moves to be merged
+        values["date_deadline"] = self._release_get_deadline(date_planned)
+        # It cannot be scheduled for now otherwise it will immediately appear
+        # in the late picking count. So it is also set to the deadline.
+        values["date_planned"] = self._release_get_deadline(date_planned)
```

Comment:
We should not change the `date_deadline` of moves (especially OUT ones), as this is the date which could have been communicated to the customer when confirming the SO (it is even enforced if we set a commitment date on the SO).
On std Odoo 14.0 the computation is the following:
```
PROC_VALUES.date_deadline = SO.commitment_date or (SO.date_order + SOL.customer_lead)
PROC_VALUES.date_planned = PROC['date_deadline'] - COMPANY.security_lead
MOVE.date = PROC_VALUES.date_planned - RULE.delay
MOVE.date_deadline = PROC_VALUES.date_deadline - RULE.delay
```
This computation has been overridden by `sale_delivery_date` (see https://github.com/OCA/sale-workflow/pull/1968/) to take into account the WH working hours, the WH cutoff time, the time windows of the customer etc.